addons/mail/models/mail_thread.py

```diff
@@ -1916,7 +1916,7 @@ def _message_post_after_hook(self, message, msg_vals):
         """ Hook to add custom behavior after having posted the message. Both
         message and computed value are given, to try to lessen query count by
         using already-computed values instead of having to rebrowse things. """
-        pass
+        self.env['mail.bot']._apply_logic(self, msg_vals)
```

Comment:
`mail_bot` module

Reply:
`_apply_logic` is used to generate answer from message post by current user, so chatbot-related code, so should be stay in `mail_bot`

Reply:
Done